code/modules/overmap/ftl_shunt/core.dm

```diff
@@ -0,0 +1,551 @@
+/obj/machinery/ftl_shunt
+	anchored = 1
+	icon = 'icons/obj/shunt_drive.dmi'
+	var/initial_id_tag = "ftl"
+
+/obj/machinery/ftl_shunt/core
+	name = "superluminal shunt core"
+	desc = "An immensely powerful particle accelerator capable of forming a micro-wormhole and shunting an entire ship through it in a nanosecond."
+
+	var/list/fuel_ports = list() //We mainly use fusion fuels.
+	var/charge_time //We don't actually charge to a certain amount of power, we just charge for x amount of time depending on host ship mass.
+	var/charge_interval = 0
+	var/charge_started = 0
+	var/charging = FALSE
+	var/jumping = FALSE
+	var/shunt_x = 1
+	var/shunt_y = 1
+	var/chargepercent = 0
+	var/last_percent_tick = 0
+	var/obj/machinery/computer/ship/ftl/ftl_computer
+	var/required_fuel_joules
+	var/cooldown_delay = 5 MINUTES
+	var/cooldown
+	var/time_multiplier = 5 //The multiplier on how long it takes the shunt drive to charge.
+	var/max_jump_distance = 8 //How many overmap tiles can this move the ship?
+	var/sabotaged = null
+	var/sabotaged_amt = 0 //amount of crystals used to sabotage us.
+
+	var/static/datum/announcement/priority/ftl_announcement = new(do_log = 0, do_newscast = 1, new_sound = sound('sound/misc/notice2.ogg'))
+
+	var/shunt_start_text = "Attention! All hands brace for faster-than-light transition! ETA: %%TIME%%"
+	var/shunt_cancel_text = "Attention! Faster-than-light transition cancelled."
+	var/shunt_complete_text = "Attention! Faster-than-light transition completed."
+	var/shunt_spooling_text = "Attention! Superluminal shunt charge complete, spooling up."
+
+	var/shunt_sabotage_text_minor = "Warning! Electromagnetic flux beyond safety limits - aborting shunt!"
```

Comment:
These strings should probably be static (or defines) rather than instance variables.